lm_reinvest.py

```diff
@@ -0,0 +1,122 @@
+log_id = ""
+
+settings_path = sys.argv[1] if len(sys.argv) > 1 else None
+# read settings
+if settings_path is not None:
+    print("Importing settings from %s" % settings_path)
+    with open(settings_path) as f:
+        settings = json.load(f)
+        utils.NODE_USER = settings['NODE_USER']
+        utils.NODE_PASSWORD = settings['NODE_PASSWORD']
+        address = settings['address']
+        min_reinvest = settings['minReinvest']
+
+        if "logToFile" in settings:
+            log_to_file = settings['logToFile']
+        if "logToConsole" in settings:
+            log_to_console = settings['logToConsole']
+        if "logId" in settings:
+            log_id = settings['logId']
+            utils.logId = log_id
+        if "telegram" in settings:
+            utils.TELEGRAM_TOKEN = settings['telegram']['token']
+            utils.TELEGRAM_CHANNEL = settings['telegram']['channel']
+
+logger = utils.setup_logger("lm-reinvest" + log_id, logging.INFO, log_to_console, log_to_file)
+utils.LOGGER = logger
+
+assets = lm_pair.split("-")
+asset_a = assets[0]
+asset_b = assets[1]
+
+utils.send_telegram(f"starting to reinvest in DFI-DUSD")
+logger.info(f"starting to reinvest in pair {asset_a}-{asset_b}. Will reinvest if DFI reaches {min_reinvest}")
+
+try:
+    while asset_a == "DUSD":
+        balance_DFI = utils.get_balance(address, "DFI")
+        if balance_DFI > min_reinvest:
+            amount_to_swap = round(balance_DFI / 2, 8)
+
+            data = {
+                "from": address,
+                "tokenFrom": asset_b,
+                "amountFrom": amount_to_swap,
+                "to": address,
+                "tokenTo": asset_a
+            }
+
+            logger.info(
+                f"{rpc('getblockcount')} initiating swap "
+                f"{round(amount_to_swap, 3)}@{asset_b}")
+            tx_id = rpc("compositeswap", [data, utils.get_tx_input(address)])
+            waitForTx(tx_id)
+            
+            blockcount = rpc('getblockcount')
+
+            # get information about pool
+            pool = utils.get_pool(lm_pair)
+
+            # get current balances for DFI
+            needed_DFI = utils.get_balance(address, "DFI")
+
+            # calculate how many DUSD are needed, based on LM-pool
+            needed_DUSD = pool['reserveA/reserveB'] * needed_DFI
+
+            # check if enough balances are around, otherwise reverse calculations
+            balance_DUSD = utils.get_balance(address, "DUSD")
+            if balance_DUSD < needed_DUSD:
+                logger.info(f"{blockcount} reversing calculations $ {balance_DUSD} < {needed_DUSD}")
+                # we need to reverse calculations, as we have not enough DUSD on our address
+                needed_DUSD = balance_DUSD
+                needed_DFI = pool['reserveB/reserveA'] * needed_DUSD
+
+                # just for safety reasons
+                balance_DFI = utils.get_balance(address, "DFI")
+                if balance_DFI < needed_DFI:
```

Comment:
will never happen. you took needed_DFI as balance_DFI first, then only reduced based on DUSD. so needed_DFI is always less than balance here